subprojects/core/src/main/java/org/gradle/api/internal/project/BuildOperationProjectConfigurator.java

```diff
@@ -0,0 +1,151 @@
+        return project;
+    }
+
+    public void runBlockConfigureClosure(BuildOperationDetails details, final Iterable<Project> projects, final Closure<? super Project> configureClosure) {
+        Action<BuildOperationContext> buildOperationAction = new Action<BuildOperationContext>() {
+            @Override
+            public void execute(BuildOperationContext buildOperationContext) {
+                for (Project project : projects) {
+                    runProjectConfigureClosure(project, configureClosure);
+                }
+            }
+        };
+        runBuildOperation(details, buildOperationAction);
+    }
+
+    public void runBlockConfigureAction(BuildOperationDetails details, final Iterable<Project> projects, final Action<? super Project> configureAction) {
+        Action<BuildOperationContext> buildOperationAction = new Action<BuildOperationContext>() {
+            @Override
+            public void execute(BuildOperationContext buildOperationContext) {
+                for (Project project : projects) {
+                    runProjectConfigureAction(project, configureAction);
+                }
+            }
+        };
+        runBuildOperation(details, buildOperationAction);
+    }
+
+    private void runProjectConfigureClosure(final Project project, final Closure<? super Project> configureClosure) {
+        Action<BuildOperationContext> buildOperationAction = new Action<BuildOperationContext>() {
+            @Override
+            public void execute(BuildOperationContext buildOperationContext) {
+                ConfigureUtil.configure(configureClosure, project);
+            }
+        };
+        runBuildOperation(computeProjectBuildOperationDetails(project), buildOperationAction);
+    }
+
+    private void runProjectConfigureAction(final Project project, final Action<? super Project> configureAction) {
+        Action<BuildOperationContext> buildOperationAction = new Action<BuildOperationContext>() {
+            @Override
+            public void execute(BuildOperationContext buildOperationContext) {
+                Actions.with(project, configureAction);
+            }
+        };
+        runBuildOperation(computeProjectBuildOperationDetails(project), buildOperationAction);
+    }
+
+    private void runBuildOperation(BuildOperationDetails details, Action<? super BuildOperationContext> buildOperationAction) {
+        buildOperationExecutor.run(details, buildOperationAction);
+    }
+
+    private BuildOperationDetails computeConfigurationBlockBuildOperationDetails(String configurationBlockName) {
+        return BuildOperationDetails.displayName("Executing '" + configurationBlockName + " {}' action").name(configurationBlockName).build();
+    }
+
+    private BuildOperationDetails computeProjectBuildOperationDetails(Project project) {
+        String name = getProjectPath(project);
+        return BuildOperationDetails.displayName("Configure " + name).name(StringUtils.capitalize(name)).build();
```

Comment:
We're doing two concatenations where one would be enough: 

`"Configure project " + project.identityPath`